#166: Add documentation to tapiro (closes #166) (#167)
buildo/retro: docs/tapiro/migrate.md

---
id: migrate
title: Migration from Wiro
---

> Tapiro is meant to deprecated [wiro](https://github.com/buildo/wiro).

Reviewer: Typo: Tapiro is meant to _deprecate_

Tapiro is based on the same concepts of wiro, the migration is pretty straightforward.

Here is a checklist of what you need to do:
1. Install the plugin (as described in the [guide](installation.md))
2. Configure your `build.sbt` (as described in the [guide](installation.md))
3. Add `AuthToken` type parameter to controllers

>     `trait AccountController` -> `trait AccountController[AuthToken]`

Reviewer: Don't controllers take 2 type parameters?

4. Modify controllers so that wiro `Auth` is replaced with AuthToken and move as last argument
    `def read(token: Auth, arg: Int)` -> `def read(arg: Int, token: AuthToken)`

> 5. Add `.gitattributes` file containing `**/*Endpoints.scala linguist-generated`

Reviewer 2: Maybe explain why?

> 5. Add `**/*Endpoints.scala linguist-generated` to repository's `.gitattributes` to hide tapiro generated code from github diff's

Reviewer 2: ```suggestion
5. Add `**/*Endpoints.scala linguist-generated` to repository's `.gitattributes` to automatically collapse tapiro generated code in GitHub diffs
```

> 5. Add `.gitattributes` file containing `**/*Endpoints.scala linguist-generated` so that autogenerated code is hidden by default in git diffs

Reviewer: I find this line hard to understand. I think it's better
> Add `**/*Endpoints.scala linguist-generated` to repository's `.gitattributes` to hide tapiro generated code from git diff's

Reviewer 2: s/git/github
